src/geocoders/mapbox.js

```diff
@@ -4,18 +4,25 @@ var L = require('leaflet'),
 module.exports = {
 	class: L.Class.extend({
 		options: {
-			serviceUrl: 'https://api.tiles.mapbox.com/v4/geocode/mapbox.places-v1/'
+			serviceUrl: 'https://api.tiles.mapbox.com/v4/geocode/mapbox.places-v1/',
+			geocodingQueryParams: {}
 		},
 
 		initialize: function(accessToken, options) {
+			var params = options.geocodingQueryParams;
+			params.access_token = accessToken;
+			if (typeof params.proximity !== 'undefined'
```

Comment:
What do you think about moving this check to the `geocode` method, so that it will work even if the options are updated after the control has been initialized?

I would thing it's quite common to set the `proximity` option to the map's center or similar, so that it will be updated as the user moves around.